### ISSUE TITLE
additional zone types

### ISSUE THREAD

**Issue by dulitz** [craigjmidwinter/total-connect-client#136]
I found [this training resource](http://techresource.online/training/ssnw/honeywell/zone-types) while searching recently. It gives very clear descriptions of each of the zone types, including rarely used ones like Zone Type 16 (which I have in my system).

**Comment by austinmroczek** [comment]
This is good info, but I think we'll need to do a more thorough test and listing of types, per panel type, and perhaps per device type.  

I have a number of devices set (in my Lynx 7000 panel) to various response types, but they report differently in TotalConnect.  I have two Entry/Exit zones but they report as type 3.  My garage door is set as Monitor but it also reports as type 3.  My keyfob reports as 3 as well.

You have one of the new panels...maybe they finally started following what the documentation says?

**Comment by dulitz** [comment]
I have two systems on Total Connect, one a year old (Vista-21IP with built-in communicator) and one about 4 years old (one of the Vista series with an iGSMV4G communicator).

My zone types 1, 3, 4, 9, 10, 16 all match precisely with the zone types as documented in the link above.

I looked at this [Lynx 7000 programming guide](https://www.alarmliquidators.com/content/5210%207000-%20Installation%20and%20Programming%20Guide.PDF) and it agrees with the link above in terms of the order of the zones, but there are no zone type numbers listed there at all.

I was thinking in terms of the constants in zone.py:
```
ZONE_TYPE_SECURITY = 0
ZONE_TYPE_LYRIC_CONTACT = 1
ZONE_TYPE_PROA7_SECURITY = 3
ZONE_TYPE_LYRIC_MOTION = 4
ZONE_TYPE_LYRIC_POLICE = 6
ZONE_TYPE_PROA7_POLICE = 7
ZONE_TYPE_FIRE_SMOKE = 9
ZONE_TYPE_PROA7_INTERIOR_DELAY = 10
ZONE_TYPE_LYRIC_TEMP = 12
ZONE_TYPE_PROA7_FLOOD = 12
ZONE_TYPE_CARBON_MONOXIDE = 14
ZONE_TYPE_PROA7_MEDICAL = 15
ZONE_TYPE_LYRIC_LOCAL_ALARM = 89
```

The docs and our constants agree that
 - zone type 9 is fire and smoke
 - zone type 14 is carbon monoxide

There is no constant for zone type 16, I have zone type 16, and it does what the docs say. I'd like to add `ZONE_TYPE_FIRE_W_VERIFICATION = 16`.

There is no constant for zone types 2, 5, or 8. I think we should add constants to match the docs.

The constants for zone type 12 imply that it's a panel-specific zone for flood or temperature, when those are monitor functions and the docs say it's a monitor zone type. I'd like to replace these with `ZONE_TYPE_MONITOR = 12`.

The constant for zone type 10 implies that it's panel specific, but the docs say that generally it's interior with delay. I'd like to replace that one with `ZONE_TYPE_INTERIOR_DELAY = 10`. Same with zone type 4; I'd like to replace that with `ZONE_TYPE_INTERIOR_FOLLOWER = 4`.

The constants for 6 and 7 imply that they are panel specific zone types for police, but the docs say that zone 6 is 24 silent and zone 7 is 24 hour audible, either of which might be used for police. I would like to replace them with `ZONE_TYPE_POLICE_SILENT = 6` and `ZONE_TYPE_POLICE_AUDIBLE = 7`.

Similarly with zone types 1 and 2.

Does your Lynx 7000 report any zones as anything other than zone type 3? Is it or its communicator perhaps just broken?

**Comment by austinmroczek** [comment]
Sorry I must have confused you with one of the other folks recently who said they had the new ProA7 panel.

My Lynx 7000 panel with ILP5 (ethernet) communicator returns 9 for fire/smoke and 14 for carbon monoxide, but for everything else it returns 3.  I haven't tried every possible combination of settings, but I have zones that should be returning 1, 2, 5 and 12 that all return 3.

Updating the constants makes sense.  We'll have to be careful in the TotalConnectZone functions.

**Comment by austinmroczek** [comment]
fixed by #143
